website/content/docs/upgrading/upgrade-specific.mdx

```diff
@@ -14,6 +14,68 @@ provided for their upgrades as a result of new features or changed behavior.
 This page is used to document those details separately from the standard
 upgrade flow.
 
+## Consul 1.10.0
+
+### Envoy xDS Protocol Upgrades
+
+Consul versions 1.9 and earlier exposed an xDS server for use by
+[Envoy](https://www.envoyproxy.io) proxies using the v2 ["State of the
+World"](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-docs/xds_protocol#variants-of-the-xds-transport-protocol)
+protocol variant.
+
+Consul 1.10.0 adds support for the v3
+[Incremental](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-docs/xds_protocol#incremental-xds)
+protocol variant as the preferred way of conversing with Envoy. Both protocol
+variants are supported in this Consul version to facilitate upgrading Consul
+and Envoy in a stairstep order to avoid downtime.
+
+In a future version of Consul the v2 State of the World protocol support will
+be removed.
+
+| Protocol           | Version | Compatible Envoy Versions | Compatible Consul Versions |
+| ------------------ | ------- | ------------------------- | -------------------------- |
+| Incremental        | v3      | 1.18.x, 1.17.x            | 1.10.x                     |
+| State of the World | v2      | 1.16.x and older          | 1.10.x and older           |
+
+#### Escape Hatches
+
+Any [escape hatches](/docs/connect/proxies/envoy#advanced-configuration) that
+are defined will likely need to be switched from using xDS v2 to xDS v3
+structures. Mostly this involves migrating off of deprecated (and now removed)
+fields and switching untyped config to [typed config](https://www.envoyproxy.io/docs/envoy/v1.17.2/configuration/overview/extension)
+with `@type` attributes set appropriately.
+
+xDS v3 syntax has been [supported since Envoy
+1.13.0](https://www.envoyproxy.io/docs/envoy/v1.13.0/api-v3/api) so this should
+be doable on most earlier versions of Consul+Envoy in advance of the Consul
+1.10.0 upgrade.
+
+As an example, here's a Zipkin integration 
+[before](https://github.com/hashicorp/consul/blob/v1.9.5/test/integration/connect/envoy/case-zipkin/service_s2.hcl)
+and [after](https://github.com/hashicorp/consul/blob/71d45a34601423abdfc0a64d44c6a55cf88fa2fc/test/integration/connect/envoy/case-zipkin/service_s2.hcl)
+
+#### Stairstep Upgrade Path
+
+1. Upgrade Envoy sidecars to the latest version of Envoy that is
+   [supported](/docs/connect/proxies/envoy#supported-versions) by the current
+   running version of Consul and 1.10.0.
+
+1. Determine if you are using the [escape hatch](/docs/connect/proxies/envoy#advanced-configuration)
+   feature. If so, rewrite the escape hatch to use the xDS v3 syntax and
+   reconfigure/restart to purge v2 elements from any configs.
```

Comment:
What's meant by reconfigure here?

Reply:
Reconfigure the escape hatch in the service registration. I am not entirely sure you'd always have to restart the envoy proxy for those to 100% pick up but I figured it wouldn't hurt to include that also.

Reply:
Rephrased